### PR TITLE
Actually install libraries for NXtranslate

### LINKED ISSUE #419: CMake doesn't install all NXtranslate libraries
When building and subsequently installing `NXtranslate` the current CMake config files do not actually install the required libraries.

### PR DESCRIPTION
**Pull request by stuartcampbell**
Added lines to `CMakeLists.txt` files to actually install the libraries

fixes #419

### CODE REVIEW

**Comment by eugenwintersberger** [comment]
@stuartcampbell According to the last telecon (May 10th 2016) the library should be installed optionally. However, if we do this we also should install the header files. Would 
`$PREFIX/include/nxtranslate` be a reasonable installation location for the header files?

**Comment by eugenwintersberger** [comment]
@stuartcampbell Is there a reason why you want to install the library in the CMake component ``Runtime``. As the libraries are all static they are not needed to run the program. If we want to install the libraries they should go to the ``Development`` component. Do I miss here something?

**Comment by stuartcampbell** [comment]
When you build using the CMake system, the libraries are not static i.e.. `.so` - so you do need them to actually run any of the tools.

**Comment by stuartcampbell** [comment]
@eugenwintersberger The libraries should be installed optionally, I agree, but that 'switch' is the CMake flag, and if it's on it should build/install all the tools.

The actual packing and installation are two different problems, if you look at the rpm spec that I have put togther then the tools are installed as part of an optional rpm called `nexus-tools`

**Comment by eugenwintersberger** [comment]
@stuartcampbell It is interesting to hear that the libraries are shared object in your case. On Linux add_library seems to build static libraries by default. I guess you do the job on OSX, am I right?
If yes, is it possible that on OSX the libraries are dynamic by default?
This raises another question: do you need the libraries to develop code or just to run nxtranslate. 
In the later case we could also use static objects and omit the libraries entirely.

**Comment by stuartcampbell** [comment]
@eugenwintersberger Nope I am building on either RHEL7 or Fedora 23

**Comment by stuartcampbell** [comment]
@eugenwintersberger And on your latest commit, you should either switch nxtranslate to be static or add the actual `nxtranslate` into your new optional flag for an optional flag - at the moment it looks like I could build the apps and then not actually have the libs to run them - which makes no sense!

File: CMakeLists.txt [hunk 1]
```diff
@@ -95,6 +95,7 @@ option(ENABLE_FORTRAN77 "Build the Fortran 77 bindings" OFF)
 option(ENABLE_JAVA      "Build Java bindings" OFF)
 option(ENABLE_CXX       "Build C++ bindings" OFF)
 option(ENABLE_APPS      "Build utility applications" OFF)
+option(ENABLE_INSTALL_NXTRANSLATE_LIBS "Install libraries for nxtranslate" OFF)
```
Comment (stuartcampbell) [inline CMakeLists.txt]: I don't understand the point of this option! you need it on if you turn ENABLE_APPS on !

**Comment by stuartcampbell** [comment]
@eugenwintersberger Looking further at your commit, I really don't understand the logic of it - Are you saying that we are treating `nxtranslate` as a development tool and shouldn't be installed ! I think you are confusing the CMAKE components and how we will actually make the packages.  @zjttoefs @peterfpeterson  any thoughts!

**Pull request merged**